docs/tutorials/acme/auto-le-cert.rst

```diff
@@ -0,0 +1,149 @@
+.. code-block:: shell
+   helm install \
+      --name cert-manager \
+      --namespace kube-system \
+     --set ingressShim.defaultIssuerName=letsencrypt-staging \
+     --set ingressShim.defaultIssuerKind=ClusterIssuer \
+     stable/cert-manager
+
+2. Creating an ACME ClusterIssuer
+=============================================================
+
+Create a file named ``cluster-issuer.yaml``:
+
+.. code-block:: yaml
+   :linenos:
+   :emphasize-lines: 11
+
+   apiVersion: certmanager.k8s.io/v1alpha1
+   kind: ClusterIssuer
+   metadata:
+     # Adjust the name here accordingly
+     name: letsencrypt-staging
+   spec:
+     acme:
+       # The ACME server URL
+       server: https://acme-staging-v02.api.letsencrypt.org/directory
+       # Email address used for ACME registration
+       email: user@example.com
+       # Name of a secret used to store the ACME account private key from step 3
+       privateKeySecretRef:
+         name: letsencrypt-private-key
+       # Enable the HTTP-01 challenge provider
+       http01: {}
+
+We then submit this file to our Kubernetes cluster:
+
+.. code-block:: shell
+
+   $ kubectl create -f cluster-issuer.yaml
+
+You should be able to verify the ACME account has been verified successfully:
+
+.. code-block:: shell
+   :emphasize-lines: 26-31
+
+   $ kubectl describe clusterissuer letsencrypt-staging
+   Name:         letsencrypt-staging
+   Namespace:
+   Labels:       <none>
+   Annotations:  <none>
+   API Version:  certmanager.k8s.io/v1alpha1
+   Kind:         ClusterIssuer
+   Metadata:
+     Cluster Name:
+     Creation Timestamp:  2017-11-30T22:33:40Z
+     Generation:          0
+     Resource Version:    4450170
+     Self Link:           /apis/certmanager.k8s.io/v1alpha1/letsencrypt-staging
+     UID:                 83d04e6b-d61e-11e7-ac26-42010a840044
+   Spec:
+     Acme:
+       Email:  user@example.com
+       Http 01:
+       Private Key Secret Ref:
+         Key:
+         Name:  letsencrypt-private-key
+       Server:  https://acme-staging-v02.api.letsencrypt.org/directory
+   Status:
+     Acme:
+       Uri:  https://acme-staging-v02.api.letsencrypt.org/acme/acct/11217539
+     Conditions:
+       Last Transition Time:  2018-04-12T17:32:30Z
+       Message:               The ACME account was registered with the ACME server
+       Reason:                ACMEAccountRegistered
+       Status:                True
+       Type:                  Ready
+
+
+3. Configure ingress with a TLS placeholder and annotaion
+=========================================================
+
+The ingress-shim watches for ingress resources with 2 conditions
+- ``kubernetes.io/tls-acme: "true"`` annotation
+- an TLS Certificate resource specified
+
+The specified Certificate resource will be overwritten, so you can generate a 
+temporary self-signed certificate using openssl to complete this setup.
+
+.. code-block:: shell
+   openssl req \
+     -newkey rsa:2048 -nodes -keyout domain.key \
+     -out domain.csr
```

Comment:
This step *shouldn't* be required 🤔 I guess Rancher may require it, and also there are a few issues with the gce ingress controller right now (#606) that would help with this, however I think we should have this in a `.. note::` block that specifically says "if you are using Rancher, or newer versions of the GCE ingress controller" or similar 😄

Reply:
Also, this creates a CSR and not a CRT right?
+ this doc doesn't explain what to do with these generated assets afterward 😄 I think moving this into a notes block, and expanding it out a bit would be great.